File: manga_backend/app/schemas/comment.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class CommentCreate(BaseModel):
    Content: str
    ChapterId: uuid.UUID | None = None
    IsSpoiler: bool = False


class CommentUpdate(BaseModel):
    Content: str


class CommentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    CommentId: uuid.UUID
    UserId: uuid.UUID
    MangaId: uuid.UUID
    Username: str | None = None
    Avatar: str | None = None
    Content: str | None = None
    IsSpoiler: bool = False
    LikeCount: int = 0
    DislikeCount: int = 0
    IsDeleted: bool | None = False
    CreatedAt: datetime | None = None
    UpdatedAt: datetime | None = None


class ReportCreate(BaseModel):
    Reason: str

File: manga_frontend/src/types/comment.ts
import type { UUID } from "./common";

export interface Comment {
  CommentId: UUID;
  UserId: UUID;
  MangaId: UUID;
  Username?: string | null;
  Avatar?: string | null;
  Content?: string | null;
  IsSpoiler: boolean;
  LikeCount: number;
  DislikeCount: number;
  IsDeleted?: boolean | null;
  CreatedAt?: string | null;
  UpdatedAt?: string | null;
}

export interface CommentCreatePayload {
  Content: string;
  ChapterId?: UUID | null;
  IsSpoiler?: boolean;
}

export interface CommentUpdatePayload {
  Content: string;
}

export interface CommentReactionResponse {
  success: boolean;
  like_count: number;
  dislike_count: number;
}

export interface ReportCreatePayload {
  Reason: string;
}
